drivers/wifi/eswifi/eswifi_offload.c

```diff
@@ -484,8 +484,10 @@ void eswifi_offload_async_msg(struct eswifi_dev *eswifi, char *msg, size_t len)
 			return;
 		}
 
-		sin_addr = &net_sin(&socket->peer_addr)->sin_addr;
+		struct sockaddr_in *peer = net_sin(&socket->peer_addr);
+		sin_addr = &peer->sin_addr;
 		memcpy(&sin_addr->s4_addr, ip, 4);
+		peer->sin_port = port;
```

Comment:
Typically the port and other values in sockaddr are in network byte order. Here the port is clearly in host byte order which looks wrong. So it looks like this should be `peer->sin_port = htons(port);` instead. You can use for example net-shell (use "net conn" command) to check what is the remote port value.